lib/icinga/notification.hpp

```diff
@@ -114,8 +120,24 @@ class Notification final : public ObjectImpl<Notification>
 	void Start(bool runtimeCreated) override;
 	void Stop(bool runtimeRemoved) override;
 
+	Array::Ptr GetTypes() const override;
+	void SetTypes(const Array::Ptr& value, bool suppress_events, const Value& cookie) override;
+
+	Array::Ptr GetStates() const override;
+	void SetStates(const Array::Ptr& value, bool suppress_events, const Value& cookie) override;
+
 private:
 	ObjectImpl<Checkable>::Ptr m_Checkable;
+	// These attributes represent the actual notification "types" and "states" attributes from the "notification.ti".
+	// However, since we want to ensure that the type and state bitsets are always in sync with those attributes,
+	// we need to override their setters, and this on the hand introduces another problem: The virtual setters are
+	// called from within the ObjectImpl<Notification> constructor, which obviously violets the C++ standard [^1].
+	// So, in order to avoid alll this kind of mess, the these two attributes are have the "no_storage" flag set, and
+	// their getters/setters are as pure virtual, which means this class has to provide the implementation of them.
```

Comment:
That comment needs a bit of proofreading :sweat_smile: 

```suggestion
	// These attributes represent the actual notification "types" and "states" attributes from the "notification.ti".
	// However, since we want to ensure that the type and state bitsets are always in sync with those attributes,
	// we need to override their setters, and this on the hand introduces another problem: The virtual setters are
	// called from within the ObjectImpl<Notification> constructor, which obviously violates the C++ standard [^1].
	// So, in order to avoid all this kind of mess, these two attributes have the "no_storage" flag set, and
	// their getters/setters are pure virtual, which means this class has to provide the implementation of them.
```